### DISCUSSION TITLE
Upgrade Your Timing Precision with ZeppTimer for ZeppOS

### DISCUSSION THREAD

**Discussion by XiaomaiTX** [zepp-health/zeppos-forum#160] [General]
[ New Development Tools ]

Introducing ZeppTimer, the accurate timer solution for the ZeppOS system!

Are you tired of dealing with timing discrepancies between your system timer and the actual time in ZeppOS? Look no further than ZeppTimer, a small and refined JavaScript class that provides an accurate timer for the ZeppOS system.

Our simple but effective approach reduces timing errors and ensures your timing needs are met. Plus, if you're migrating from ZeppOS 1.0 to 2.x or above, using the ZeppTimer library can save you the cost of migration by replacing the official approach to timer components with setTimeOut and cleanTimeOut.

Say goodbye to unreliable timing and hello to the precision of ZeppTimer. Download now and see the difference for yourself.

https://github.com/XiaomaiTX/zeppos-timer

**Comment by kartun83** [comment]
So, which approach is recommended for 2.x ?
directly via setInterval or using this library ?
For 2.x documentation there is no information related to timers.

**Reply by XiaomaiTX** [reply]
This library is designed to fix some bugs in the official timer-related interface in ZeppOS, so I recommend using this library

V2.x actually splits Timer into more fine-grained interfaces, but it doesn't solve the problem of running errors

**Reply by XiaomaiTX** [reply]
The runtime error I'm referring to is a timing error

After the official timer has been running for about 30 seconds, there will probably be a 0.5 second error

I'm sorry, my English is not very good
